arkouda/segarray.py

```diff
@@ -115,12 +115,22 @@ def __init__(self, segments, values, copy=False, lengths=None, grouping=None):
         else:
             self.lengths = lengths
         self.dtype = values.dtype
+
+        self.non_empty = zeros(self.size, dtype=akbool)
+        self.non_empty_count = 0
+        for i in range(self.lengths.size):
+            if self.lengths[i] == 0:
+                self.non_empty[i] = False
+            else:
+                self.non_empty[i] = True
+                self.non_empty_count += 1
+
```

Comment:
I think we might be able to get a cleaner initialization if we do something like this

```python
self.non_empty = self.lengths > 0
self.non_empty_count = self.non_empty.sum()
```
This way we don't have to iterate through `self.lengths` on the client

Reply:
I agree. This has been updated.